src/test/java/com/salesforce/dynamodbv2/ScanTest.java

```diff
@@ -135,6 +138,45 @@ void scanAll(TestArgument testArgument) {
         });
     }
 
+
+    @ParameterizedTest(name = "{arguments}")
+    @ArgumentsSource(DefaultArgumentProvider.class)
+    void scanAllTenants(TestArgument testArgument) {
+        MT_CONTEXT.setContext(null);
+        String startTableName = null;
+        List<String> tableNames;
+        int maxPages = 5;
+        int attemptCount = 0;
+        do {
+            List<String> allTableNames = testArgument.getAmazonDynamoDb().listTables(startTableName).getTableNames();
+
+            //filter scan to just tables this mt dymamo strategy manages
+            tableNames = allTableNames.stream()
+                .filter(t -> ((MtAmazonDynamoDbBase) testArgument.getAmazonDynamoDb()).isMtTable(t))
```

Comment:
Seems like `listTables` should do this filtering internally (and then maybe we don't not need to make `isMtTable` public).